source/elements/oneDAL/source/common_interface/header_files_and_namespaces.rst

```diff
@@ -1,3 +1,61 @@
-===========================
-Header files and namespaces
-===========================
+============
+Header files
+============
+
+oneDAL public identifiers are represented in the following header files:
+
+.. list-table::
+   :widths:  10 50
+   :header-rows: 1
+
+   * - header file
+     - description
+   * - ``onedal/onedal.hpp```
+     - Main header file of oneDAL library.
+   * - ``onedal/algo/%ALGORITHM%/%ALGORITM%.hpp``
+     - An algorithm header file for a particular algorithm.
+       The string %ALGORITHM% should be substituted with the name of the algorithm, for example, ``kmeans`` or ``knn``.
+
+   * - ``onedal/algo/misc/%FUNCTIONALITY%/%FUNCTIONALITY%.hpp``
+     -  A header file for miscellaneous data types and functionality intended to be used by oneDAL algorithms and applications for algorithm customization and optimization on various stages
+        of the analytical pipeline.
+        The string %FUNCTIONALITY% should be substituted with the functionality name, for example, ``mt19937`` or ``cross_enthropy_loss``.
+
+   * - ``onedal/util/%UTILITY%.hpp```
+     - A header file for auxiliary functionality intended to be used for design of classes and implementation of methods such as memory allocators or type traits. 
+       The string %UTILITY% should be substituted with the auxiliary functionality name, for example, ``usm_allocator`` or ``type_traits``. 
```

Comment:
q: do you need to have ``onedal/`` part for each header file? Why not mention before the table that all of these files are inside the ``ondeal`` directory?

concern: for the third row the path to the headerfile is too long, that causes the second column to be really small (sphinx does not break the inline code into multiple lines)
![image](https://user-images.githubusercontent.com/17865025/83018294-b6426b00-a02d-11ea-96a4-38fbe9f7b36c.png)

Reply:
the solution I can think of is to break the path into the directory and the name of the file, and have something like this:
```
filename: ``%ALGORITM%.hpp``

located in: ``onedal/algo/%ALGORITHM%/``
```

not sure it's a good idea

Reply:
From my point of view, it is better to mention "onedal" folder before the table and omit it inside the table. Option with filename and location is also ok for me